src/synthetic_dataset_generator/apps/rag.py

```diff
@@ -0,0 +1,918 @@
+                with gr.Tab("Load your file") as tab_file_input:
+                    with gr.Row(equal_height=False):
+                        with gr.Column(scale=2):
+                            file_in = gr.File(
+                                file_count="multiple",
+                                label="Upload your file",
+                                file_types=[".md", ".txt"],
+                            )
+                            with gr.Row():
+                                clear_file_btn_part = gr.Button(
+                                    "Clear", variant="secondary"
+                                )
+                                load_file_btn = gr.Button("Load", variant="primary")
+                        with gr.Column(scale=3):
+                            file_out = gr.HTML(label="Dataset preview", visible=False)
+                with gr.Tab("Generate from prompt") as tab_prompt_input:
+                    with gr.Row(equal_height=False):
+                        with gr.Column(scale=2):
+                            dataset_description = gr.Textbox(
+                                label="Dataset description",
+                                placeholder="Give a precise description of your desired dataset.",
+                            )
+                            with gr.Row():
+                                clear_prompt_btn_part = gr.Button(
+                                    "Clear", variant="secondary"
+                                )
+                                load_prompt_btn = gr.Button("Create", variant="primary")
+                        with gr.Column(scale=3):
+                            examples = gr.Examples(
+                                examples=DEFAULT_DATASET_DESCRIPTIONS,
+                                inputs=[dataset_description],
+                                cache_examples=False,
+                                label="Examples",
+                            )
+
+        gr.HTML(value="<hr>")
+        gr.Markdown(value="## 2. Configure your task")
+        with gr.Row(equal_height=True):
+            with gr.Row(equal_height=False):
+                with gr.Column(scale=2):
+                    system_prompt = gr.Textbox(
+                        label="System prompt",
+                        placeholder="You are a helpful assistant.",
+                        visible=False,
+                    )
+                    document_column = gr.Dropdown(
+                        label="Document Column",
+                        info="Select the document column to generate the RAG dataset",
+                        choices=["Load your data first in step 1."],
+                        value="Load your data first in step 1.",
+                        interactive=False,
+                        multiselect=False,
+                        allow_custom_value=False,
+                    )
+                    hard_negative = gr.Checkbox(
```

Comment:
perhaps it is more intuitive to combine these options in a single checkboxgroup? For me at first it was not super clear what the difference was between not selecting them and selecting them, and whether or not they were going to be exclusive or extra.